test/printf-test.cc

```diff
@@ -554,3 +554,51 @@ TEST(PrintfTest, VSPrintfMakeWArgsExample) {
                           fmt::make_wprintf_args(42, L"something")));
 #endif
 }
+
+typedef fmt::printf_arg_formatter<
+    fmt::back_insert_range<fmt::internal::buffer<char>>> formatter_t;
+typedef fmt::basic_printf_context<formatter_t::iterator, char> context_t;
+
+// A custom printf argument formatter that doesn't print `-` for floating-point
+// values rounded to 0.
+class custom_printf_arg_formatter : public formatter_t {
+ public:
+    using formatter_t::iterator;
+
+  custom_printf_arg_formatter(formatter_t::iterator iter, formatter_t::format_specs& spec, context_t& ctx)
+      : formatter_t(iter, spec, ctx) {}
+
+  using formatter_t::operator();
+
+#if FMT_MSC_VER > 0 && FMT_MSC_VER <= 1804
+      template <typename T, FMT_ENABLE_IF(std::is_floating_point<T>::value)>
+      iterator operator()(T value) {
+#else
```

Comment:
Can we use the above SFINAE-based definition for all compilers and remove conditional compilation (adding a comment clarifying that this is a workaround for MSVC).

Reply:
Unfortunately this doesn't work. Other compilers (clang and gcc as far as I recall) complain then.

Reply:
OK, let's keep it as is.